### PR TITLE
vecsim query results api + gtest

### PR DESCRIPTION
**Pull request by DvirDukhan**
This PR introduces `VecSimQueryResult` API
`VecSimQueryResult_Len` - returns the actual length of the query result array
`VecSimQueryResult_Free` - free query result array

build & Test change:
use gtest instead of redismodule

### CODE REVIEW

**Review by ashtul — COMMENTED** [review]
@rafie can you take a look at all the build changes
Small comments otherwise look good. :+1:

File: tests/flow/test_llapi.py [hunk 1]
```diff
@@ -10,39 +10,40 @@
 def test_llapi_hnswlib_vector_add(env):
     con = env.getConnection()
     res = con.execute_command("vec_sim_test.hnswlib_vector_add")
-    env.assertEquals(res, "OK")
+    env.assertEqual(res, "OK")
```
[the remark this answers is not in the document]
Comment (DvirDukhan) [reply tests/flow/test_llapi.py:13]: thanks.
note that currently, the python Redis integration tests are disabled, as we moved to test for LLAPI test.
I plan to have Redis integration tests different

[the remark this answers is not in the document]
Comment (ashtul) [reply tests/flow/test_llapi.py:13]: This will simply save a line of code and make it easier to read.

**Review by ashtul — COMMENTED** [review]
LGTM.

**Review by rafie — APPROVED** [review]

**Pull request merged**